### PR TITLE
chart: incrementals-publisher unit tests for secrets

### PR DESCRIPTION
**Pull request by SarathChandra24**
Unit tests for verifying secrets values and their assignment to deployment env.

I believe this is backward compatible as only the unit tests and values are changes.

```sh
helm template incrementals-publisher 
```

returns the following 
```YAML
---
# Source: incrementals-publisher/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: release-name-incrementals-publisher
  labels:
    app.kubernetes.io/name: incrementals-publisher
    helm.sh/chart: incrementals-publisher-0.7.0
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
type: Opaque
data:
  artifactory_key: dGVzdGluZ0FydGlmYWN0b3J5S2V5
  github_token: Y3VzdG9tU3VwZXJTZWNyZXRUb2tlbg==
  github_app_id: Y3VzdG9tU3VwZXJTZWNyZXRBcHBJZA==
  github_app_private_key: Y3VzdG9tU3VwZXJTZWNyZXRQcml2YXRlS2V5
  jenkins_auth: Y3VzdG9tU2VjcmV0SmVua2luc1VzZXI6Y3VzdG9tU2VjcmV0SmVua2luc1Bhc3N3b3Jk
  preshared_key: Y3VzdG9tU3VwZXJTZWNyZXRQcmVTaGFyZWRLZXk=
---
# Source: incrementals-publisher/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-incrementals-publisher
  labels:
    app.kubernetes.io/name: incrementals-publisher
    helm.sh/chart: incrementals-publisher-0.7.0
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
spec:
  type: ClusterIP
  ports:
    - port: 8080
      targetPort: http
      protocol: TCP
      name: http
  selector:
    app.kubernetes.io/name: incrementals-publisher
    app.kubernetes.io/instance: release-name
---
# Source: incrementals-publisher/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-name-incrementals-publisher
  labels:
    app.kubernetes.io/name: incrementals-publisher
    helm.sh/chart: incrementals-publisher-0.7.0
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: incrementals-publisher
      app.kubernetes.io/instance: release-name
  template:
    metadata:
      annotations:
        # subpath secrets don't get updates, and we want to restart on change anyways
        checksum/secret: d1b352b96ce0304125d13f2f6163589249ab793e8c051f8f2ba51347ff19febf
      labels:
        app.kubernetes.io/name: incrementals-publisher
        app.kubernetes.io/instance: release-name
    spec:
      automountServiceAccountToken: false
      securityContext:
        {}
      containers:
        - name: incrementals-publisher
          securityContext:
            {}
          image: "jenkinsciinfra/incrementals-publisher:1.8.10"
          imagePullPolicy: 
          ports:
            - name: http
              containerPort: 3000
              protocol: TCP
          #livenessProbe:
          #  httpGet:
          #    path: /healthcheck
          #    port: http
          #readinessProbe:
          #  httpGet:
          #    path: /healthcheck
          #    port: http
          resources:
            {}
          env:
            - name: JENKINS_AUTH
              valueFrom:
                secretKeyRef:
                  name: release-name-incrementals-publisher
                  key: jenkins_auth
            - name: GITHUB_TOKEN
              valueFrom:
                secretKeyRef:
                  name: release-name-incrementals-publisher
                  key: github_token
            - name: GITHUB_APP_ID
              valueFrom:
                secretKeyRef:
                  name: release-name-incrementals-publisher
                  key: github_app_id
            - name: GITHUB_APP_PRIVATE_KEY
              valueFrom:
                secretKeyRef:
                  name: release-name-incrementals-publisher
                  key: github_app_private_key
            - name: ARTIFACTORY_KEY
              valueFrom:
                secretKeyRef:
                  name: release-name-incrementals-publisher
                  key: artifactory_key
            - name: PRESHARED_KEY
              valueFrom:
                secretKeyRef:
                  name: release-name-incrementals-publisher
                  key: preshared_key
```

https://github.com/jenkins-infra/helm-charts/issues/656

### CODE REVIEW

**Comment by SarathChandra24** [comment]
I specified the issue in the comment above at the end. May be it is not visible. Here is the issue to which this PR is linked.
https://github.com/jenkins-infra/helm-charts/issues/656

I will forward the discussion to the issue.

**Comment by dduportal** [comment]
> I specified the issue in the comment above at the end. May be it is not visible. Here is the issue to which this PR is linked. #656
> 
> I will forward the discussion to the issue.

My apologies, I missed the link to the issue which was the last item in the body. Thanks for the reminder. Let's continue in the issue then!

**Review by dduportal — COMMENTED** [review]
Side note: you might want to take care of following the POSIX rules regarding end of files.

The current git diff shows no empty line at the end of files:

Capture d’écran 2024-02-21 à 09 28 22

See https://thoughtbot.com/blog/no-newline-at-end-of-file to have more informations. Usually you can set your IDE/text editor to automatically add these end of file empty characters

File: charts/incrementals-publisher/values.yaml [hunk 1]
```diff
@@ -50,12 +50,12 @@ affinity: {}
 permissions_url: ''
 artifactory:
   # artifactory.key -- key to upload to artifactory
-  key: ''
+  key: ""
```
Comment (dduportal) [inline charts/incrementals-publisher/values.yaml:53]: Is there a particular reason to change this? Could you keep the original simple quote under the proposed argument to stay coherent within the file?

File: charts/incrementals-publisher/tests/defaults_test.yaml [hunk 2]
```diff
@@ -2,10 +2,15 @@ suite: Default values
 templates:
   - deployment.yaml
   - secret.yaml # dependency of deployment.yaml
-values:
-  # These values are required by the template "secret.yaml"
-  - secret-values.yaml
+# values:
+#   # These values are required by the template "secret.yaml"
+#   - secret-values.yaml
```
Comment (dduportal) [inline charts/incrementals-publisher/tests/defaults_test.yaml:7]:
```suggestion
```

Comment (dduportal) [reply charts/incrementals-publisher/tests/defaults_test.yaml:7]: Required: leftover comments

File: charts/incrementals-publisher/tests/defaults_test.yaml [hunk 3]
```diff
@@ -2,10 +2,15 @@ suite: Default values
...
 tests:
+  - it: should create an secret with default values
```
Comment (dduportal) [inline charts/incrementals-publisher/tests/defaults_test.yaml:9]:
```suggestion
  - it: should create a secret with default values
```

nit: typo

File: charts/incrementals-publisher/tests/custom_values_test.yaml [hunk 4]
```diff
@@ -2,11 +2,17 @@ suite: Custom values
...
   permissions_url: https://somewhere.jenkins.io/github.json
+  artifactory:
+    key: testingArtifactoryKey
+  github:
+      token: customSuperSecretToken
+      appId: customSuperSecretAppId
+      privateKey: customSuperSecretPrivateKey
+  jenkins:
+    auth: customSecretJenkinsUser:customSecretJenkinsPassword
+  preshared_key: customSuperSecretPreSharedKey
```
Comment (dduportal) [inline charts/incrementals-publisher/tests/custom_values_test.yaml:15]: 👏  that looks really good!

**Review by dduportal — CHANGES_REQUESTED** [review]
This new version looks really good!

I've added suggestions (2 are blocking + 1 typo) but it looks almost ready!

File: charts/incrementals-publisher/values.yaml (hunk 1 above)
Comment (SarathChandra24) [reply charts/incrementals-publisher/values.yaml:53]: Just wanted to make it consistent across the file, but will rollback as other files are also having the same.

```yaml
nameOverride: ""
fullnameOverride: ""
```

**Review by dduportal — APPROVED** [review]
LGTM, thanks!

**Pull request merged**